webanno-api-dao/src/main/java/de/tudarmstadt/ukp/clarin/webanno/api/dao/DocumentServiceImpl.java

```diff
@@ -1012,4 +1012,22 @@ public void onBeforeDocumentRemovedEvent(BeforeDocumentRemovedEvent aEvent)
     {
         projectService.recalculateProjectState(aEvent.getDocument().getProject());
     }
+
+    @Override
+    public long getNumOfSourceDocuments()
+    {
+        String query = String.join("\n",
+                "SELECT COUNT(*)",
+                "FROM SourceDocument");
+        return (long) entityManager.createQuery(query).getSingleResult();
```

Comment:
Better use `long count = entityManager.createQuery(query, Long.class)` (also in the other cases).